lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -526,6 +526,19 @@ public abstract MergeSpecification findForcedMerges(
   public abstract MergeSpecification findForcedDeletesMerges(
       SegmentInfos segmentInfos, MergeContext mergeContext) throws IOException;
 
+  /**
+   * Identifies merges that we want to execute (synchronously) on commit. By default, do not synchronously merge on commit.
```

Comment:
Can you describe a bit more?  E.g. explain that any merges returned here which then finish within the allotted time will be reflected in the commit, so that the number of segments in the commit point may be decreased?

Reply:
While updating the comments here, it occurred to me that if we want to leverage this same method in future to trigger a merge on NRT refresh, would it make sense to call it e.g. `findFlushMerges` or `findFullFlushMerges` instead?

Reply:
Good thinking!  +1 for `findFullFlushMerges`, to make it clear it's a FULL flush (`commit` or `refresh`) and not an ordinary flush e.g. when the in-memory segments are taking too much memory and some are flushed.

And then I guess the method could look at `MergeTrigger` to differentiate whether it's a `commit` or a `refresh`?

Reply:
Renamed in latest commit and added a new MergeTrigger.

Passing the `MergeTrigger` to the `MergeScheduler` got me thinking that it might be possible to raise the priority of these full-flush merges, though it would be easier to manage that in the call to `registerMerge`, which could insert them at the front of the pending merge queue. I think that's beyond the scope of this change, but it could make for an interesting follow-up.